packages/fxa-settings/src/pages/Signin/index.tsx

```diff
@@ -230,11 +231,15 @@ const Signin = ({
                 setSigninLoading(false);
                 break;
               }
+
+              // Fallback to using v1 creds for signin unblock and don't
+              // upgrade user to v2 keys
+              const v1Credentials = await getCredentials(email, password);
```

Comment:
I have a minor change request here. We should respect the current credential version the user has. i.e. If the user has an account with v2 credentials we should send v2 credentials over in `authPW`.

Doing this should be pretty straightforward. You'll want to, use the GQL `CREDENTIAL_STATUS_MUTATION`, to check the user’s credential `version` and `client_salt`. Then based on whether they have v1 or v2 you can call `getCredentials(email,password)` or `getCredentialsV2(client_salt, password)` accordingly.  Then you can send over `authPW` and the `authPWVersion` over. 

Note, that for the `BEGIN_SIGNIN_MUTATION` the `authPW` field of `signInInput` works with either a v1 or v2 stretched password.

Reply:
Thanks @dschom, I'll file a follow up since there is some more complexity around v1 creds with [email case](https://github.com/mozilla/fxa/pull/16968) and possibly rate limiting with `getCredentialStatus` while the user is already rate limited with unblock code.